pkg/controller/compliancescan/config.go

```diff
@@ -258,6 +263,15 @@ func commonOpenScapEnvCm(name string, scan *compv1alpha1.ComplianceScan) *corev1
 	return cm
 }
 
+func getOscapDebugLevel() string {
+	switch envVar := os.Getenv("OSCAP_DEBUG_LEVEL"); envVar {
+	case "DEVEL", "INFO", "WARNING", "ERROR":
+		return envVar
+	}
```

Comment:
I looked through the `oscap` manual, but didn't see a documented default value.

If we decided to use verbosity regardless of the `oscap` default, we could simplify the calling code to handle returning the appropriate debug level (without the callers having to be invoked in a particular order to maintain precedence).

```go
func getLogLevel(s *scan) string {
    switch envVar := os.Getenv("OSCAP_DEBUG_LEVEL"); envVar {
    case "DEVEL", "INFO", "WARNING", "ERROR":
        return envVar
    default:
        return "WARNING"  // is the the default we want to assume in all cases?
                          // Thinking about this in the case where someone doesn't supply
                          // a valid argument.
    }

    if scan.Spec.Debug {
        return "INFO"
    }
}
```

Then the calling code would be:

```go
cm.Data[OpenScapVerbositoryEnvName] := getLogLevel(scan)
```

Reply:
The default seems to be warning: https://github.com/OpenSCAP/openscap/blob/94cd7978e04236c75a1d7454703664f20db50a41/src/common/debug.c#L139
I'm going to use your approach, thanks!